File: 643-maximum-average-subarray-i/maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int wsum=0;
        int msum=0;
        for(int i=0;i<k;i++){
            wsum+=nums[i];
        }
        msum=wsum;
        for(int j=k;j<nums.length;j++){
            wsum=wsum-nums[j-k]+nums[j];
           
            msum=Math.max(msum,wsum);

        }
        return (double)msum/k;
    }
}

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.py
class Solution(object):
    def findMaxAverage(self, nums, k):
        sum=0;
        msum=0;
        for i in range(0,k):
            sum+=nums[i]
        msum=sum
        for i in range(k,len(nums)):
            sum=sum-nums[i-k]+nums[i]
            if(sum>msum):
                msum=sum
        return float(msum)/k


        